gcp-secret-manager/src/main/java/io/micronaut/gcp/secretmanager/client/DefaultLocationSecretManagerClient.java

```
@@ -0,0 +1,100 @@
+        this.executorService = executorService != null ? executorService : Executors.newSingleThreadExecutor()  ;
+        this.configurationProperties = configurationProperties;
+    }
+
+    @Override
+    public Mono<VersionedSecret> getSecret(String secretId) {
+        return getSecret(secretId, LATEST , googleCloudConfiguration.getProjectId());
+    }
+
+    @Override
+    public Mono<VersionedSecret> getSecret(String secretId, String version) {
+        return getSecret(secretId, version, googleCloudConfiguration.getProjectId());
+    }
+
+    @Override
+    public Mono<VersionedSecret> getSecret(String secretId, String version, String projectId) {
+        String location = configurationProperties.getLocation();
+        LOGGER.debug("Fetching secret: projects/{}/locations/{}/secrets/{}/{}", projectId, location, secretId, version);
+        SecretVersionName secretVersionName = SecretVersionName.ofProjectLocationSecretSecretVersionName(projectId, location, secretId, version);
+        AccessSecretVersionRequest request = AccessSecretVersionRequest.newBuilder()
+                .setName(secretVersionName.toString())
+                .build();
+        final Mono<AccessSecretVersionResponse> mono = Mono.create((sink) -> {
+            final ApiFuture<AccessSecretVersionResponse> future = client
+                    .accessSecretVersionCallable().futureCall(request);
+            future.addListener(() -> {
+                try {
+                    final AccessSecretVersionResponse result = future.get();
+                    sink.success(result);
+                } catch (Throwable e) {
+                    sink.error(e);
+                }
+            }, executorService);
+        });
+
+        return mono
+                .map(response -> new VersionedSecret(secretId, projectId, version, response.getPayload().getData().toByteArray(), location))
+                .onErrorResume(throwable -> Mono.empty());
```

don't just swallow the exception but add some debug logging here

---

Done